src/screens/HyperSwitch/PaymentLogs/PaymentLogs.res

```diff
@@ -323,139 +362,165 @@ let make = (~paymentId, ~createdAt) => {
           eventDict->Dict.set("created_at", timestamp->Js.Json.string)
           eventDict->Js.Json.object_
         })
-      setSdkLogsData(_ =>
-        sdkLogsArray->Array.filter(sdkLog => {
-          let eventDict = sdkLog->getDictFromJsonObject
-          let eventName = eventDict->getString("event_name", "")
-          let filteredEventNames = ["StripeElementsCalled"]
-          filteredEventNames->Array.includes(eventName)->not
-        })
-      )
-
-      // setting initial data
-      let initialData =
-        sdkLogsArray
-        ->Belt.Array.get(0)
-        ->Belt.Option.getWithDefault(Js.Json.null)
-        ->getDictFromJsonObject
-      let intialValueRequest = initialData->getString("event_name", "")
-
-      let intialValueResponse = initialData->getString("response", "")
-      setRequestObject(_ => intialValueRequest)
-      setResponseObject(_ => intialValueResponse)
-      setCurrentSelected(_ => initialData->getString("event_id", ""))
+      let logsArr = sdkLogsArray->Array.filter(sdkLog => {
+        let eventDict = sdkLog->getDictFromJsonObject
+        let eventName = eventDict->getString("event_name", "")
+        let filteredEventNames = ["StripeElementsCalled"]
+        filteredEventNames->Array.includes(eventName)->not
+      })
+      logs.contents = logs.contents->Array.concat(logsArr)
 
-      setScreenState2(_ => PageLoaderWrapper.Success)
+      PageLoaderWrapper.Success
     } catch {
     | Js.Exn.Error(e) =>
       let err = Js.Exn.message(e)->Belt.Option.getWithDefault("Failed to Fetch!")
-      setScreenState2(_ => PageLoaderWrapper.Error(err))
+      PageLoaderWrapper.Error(err)
     }
   }
+
   let getDetails = async () => {
-    try {
-      let _ = await Window.connectorWasmInit()
-      if !(paymentId->HSwitchOrderUtils.isTestPayment) {
-        fetchPaymentLogsData()->ignore
-        fetchSdkLogsData()->ignore
-      } else {
-        setScreenState1(_ => PageLoaderWrapper.Success)
-        setScreenState2(_ => PageLoaderWrapper.Success)
+    if !(paymentId->HSwitchOrderUtils.isTestPayment) {
+      let screenState = switch (
+        await fetchPaymentLogsData(),
+        await fetchSdkLogsData(),
+        await fetchWebhooksLogsData(),
+      ) {
```

Comment:
Can you define variable of these and then write switch case for that - 

`  await fetchPaymentLogsData(),
        await fetchSdkLogsData(),
        await fetchWebhooksLogsData(),`
        
Better for code readability and understanding

Reply:
not needed